src/core/shards/casting.hpp

```diff
@@ -249,7 +249,7 @@ template <SHType ToType> struct MakeVector {
     static Parameters params = []() {
       Parameters params;
       auto vectorType = VectorTypeLookup::getInstance().get(ToType);
-      auto paramType = getCompatibleUnitType(vectorType->numberType);
+      const Type &paramType = getCompatibleUnitType(vectorType->numberType);
```

Comment:
ouch

Reply:
Just curious: what wouldn't `auto` grab the correct type here? Or should have it been `auto&`?

Reply:
auto always tries to remove reference types, so you explicitly need to use auto& when you want to store a reference.
(doesn't remove pointers)

Reply:
I think const carries over, so you can assign a const reference to `auto& something = ...`